requirements.txt

```diff
@@ -9,9 +9,11 @@ numpy==1.20.1
 plaidml-keras==0.7.0
 psutil==5.8.0
 pyserial==3.5
+python-socketio==5.3.0
 python-gdcm==3.0.8.1
 scikit-image==0.18.1
 scipy==1.6.1
+requests==2.26.0
```

Comment:
You are adding requests because of another PR you are going to submit?

Reply:
Thanks for the comment! The problem here is that just installing python-socketio does not install all the required dependencies, notably, the requests package. 

However, as the following page describes, installing python-socketio[client] actually does install the dependencies:

https://github.com/miguelgrinberg/python-socketio/issues/356

Therefore, I changed python-socketio to python-socketio[client], so explicit installation of 'requests' is not needed anymore.

Preparing for the MIDI device change, I also moved the socket-io dependency to `optional-requirements.txt`, as described, e.g., here: https://stackoverflow.com/questions/3664478/optional-dependencies-in-a-pip-requirements-file

Also the requirements needed by the MIDI change can then be marked as optional by including them in the that file. (You proposed naming the file, e.g., `navigator_requirements.txt` in the other PR. I considered that, but ultimately thought that maybe it's better to have the word `optional` there, as they aren't strictly necessary for navigation, either.)